server.go

```diff
@@ -64,12 +64,17 @@ func handleComments(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		// Write out the json response
-		commentsEncoder := json.NewEncoder(cFile)
-		commentsEncoder.Encode(comments)
+		// Pretty print the JSON, write to file
+		var encodedComments, _ = json.MarshalIndent(comments, "", "    ")
```

Comment:
1. What's the difference? (I totally missed that)
2. I usually agree. We've been pretty lazy in our other servers but I could handle it.
3. It gets GCed right? It's not a huge deal but I wanted the file pretty-printed (since we talk about manually editing the file in the tutorial). I didn't see how to use `MarshalIndent` with `Encode`. It also felt a bit silly to write the file and then immediately read it (my first pass was to just `io.Copy(w, cFile)` like in the GET handler).

Like I said, I haven't written Go. If we could get this pedantically idiomatic and keep the JSON formatting that would be awesome! Either way, nobody should be taking these servers and putting them into production ;) , so it's not the end of the world if we don't get it perfect.